container/src/lib.rs

```diff
@@ -21,7 +23,18 @@ pub enum Error {
     ContainerSpawnCommand(unshare::Error),
     ContainerWaitCommand(std::io::Error),
     ContainerExit(i32),
+    ContainerExists(String),
     Unmount(std::io::Error),
+    /// Fail to save container state file.
+    WriteStateFile(String),
+    /// Fail to serialize container state file.
+    SerializeState(String),
+    /// Fail to open container state file.
+    OpenStateFile(String),
+    /// Fail to create container state file.
+    CreateStateFile(String),
+    /// Fail to remove container state file.
+    RemoveStateFile(String),
```

Comment:
As commented above, your errors should take the native type as an input, not a string.